EasyImgur/FileHelper.cs

```diff
@@ -81,5 +83,50 @@ public static byte[] PeekBytes(this Stream stream, int count)
             stream.Seek(0, SeekOrigin.Begin);
             return buffer;
         }
+
+        public static bool CheckIfValidImageFileHeader(string path)
+        {
+            using (var fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, FileOptions.SequentialScan))
+            {
+                var bytes = fs.PeekBytes(4);
+                var imgFormatFromHeader = GetImageFormat(bytes);
+                return imgFormatFromHeader != null;
+            }
+        }
+
+        public static System.Drawing.Imaging.ImageFormat GetImageFormat(byte[] bytes)
+        {
+            // see http://www.mikekunz.com/image_file_header.html  
+            var bmp = Encoding.ASCII.GetBytes("BM");     // BMP
+            var gif = Encoding.ASCII.GetBytes("GIF");    // GIF
+            var png = new byte[] { 137, 80, 78, 71 };    // PNG
+            var tiff = new byte[] { 73, 73, 42 };         // TIFF
+            var tiff2 = new byte[] { 77, 77, 42 };         // TIFF
+            var jpeg = new byte[] { 255, 216, 255, 224 }; // jpeg
+            var jpeg2 = new byte[] { 255, 216, 255, 225 }; // jpeg canon
```

Comment:
You're right, checking headers doesn't guarantee the file is a valid image.
Reason I added this is because there is a chance none of the image are valid, but at that moment we has already created album (unless we delete the album afterwards), result in empty album.  

But this reason is less valid after I added another checking using flag `hasAtLeastOneFullyValidatedImageFile`, so I'd say it's reasonable to skip checking headers.